### PR TITLE
[Function][Vectorized] Implement random/rand vectorized function

### PR DESCRIPTION
**Pull request by xinghuayu007**
## Proposed changes

Implement random/rand vectorized function.

Currently seed is ignored, rand(seed) is the same as rand(), because this function will be called multiple times for one query, rand(seed) will return duplicated results. 

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/random.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    Status execute_impl(Block &block, const ColumnNumbers &arguments, size_t result,
+                        size_t input_rows_count) override {
+        static const double min = 0.0;
+        static const double max = 1.0;
+        auto res_column = ColumnFloat64::create(input_rows_count);
+        auto& res_data = assert_cast<ColumnFloat64&>(*res_column).get_data();
+
+        // random()/rand() without seed
+        if (arguments.size() == 0) {
+            std::mt19937 generator(std::random_device{}());
```
Comment (BiteTheDDDDt) [inline be/src/vec/functions/random.h:50]:
```suggestion
            std::mt19937_64 generator(std::random_device{}());
```

**Review by BiteTheDDDDt — CHANGES_REQUESTED** [review]

File: be/src/vec/functions/random.h (hunk 1 above)
Comment (xinghuayu007) [reply be/src/vec/functions/random.h]: done

File: be/src/vec/functions/random.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+class Random : public IFunction {
+public:
+    static constexpr auto name = "random";
+
+    static FunctionPtr create() { return std::make_shared<Random>(); }
+
+    String get_name() const override { return name; }
+
+    size_t get_number_of_arguments() const override { return 1; }
+
```
Comment (zhangstar333) [inline be/src/vec/functions/random.h:36]: 无参数时，过不了，可以加上这句
bool is_variadic() const override { return true; }

Comment (xinghuayu007) [reply be/src/vec/functions/random.h:36]: 无参数场景，我测试过，是可以的

Comment (zhangstar333) [reply be/src/vec/functions/random.h:36]: 在开向量化下吗？

Comment (xinghuayu007) [reply be/src/vec/functions/random.h:36]: 对

Comment (zhangstar333) [reply be/src/vec/functions/random.h:36]: 刚试了下,Debug模式会core，Release下没有问题，但是不知道为啥无参数时返回Empty Set

Comment (xinghuayu007) [reply be/src/vec/functions/random.h:36]: 2021-09-30 09-02-53屏幕截图

Comment (HappenLee) [reply be/src/vec/functions/random.h:36]: 辛苦补充一下ut，在确定种子的值时候

File: be/src/vec/functions/random.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    String get_name() const override { return name; }
+
+    bool use_default_implementation_for_constants() const override { return false; }
+
+    size_t get_number_of_arguments() const override { return 0; }
+
+    bool is_variadic() const override { return true; }
+
+    bool use_default_implementation_for_nulls() const override { return false; }
```
Comment (HappenLee) [inline be/src/vec/functions/random.h:43]: 为什么不用默认的null实现，rand(null)应该返回null吧

**Pull request closed without merging**